doc/manual/environment.rst

```diff
@@ -1,53 +1,96 @@
 The environment
 ===============
 
-Experiments interact with an environment that consists of devices, arguments and datasets. Access to the environment is handled by the class :class:`artiq.language.environment.EnvExperiment` that experiments should derive from.
+ARTIQ experiments exist in an environment, which consists of devices, arguments, and datasets. Access to the environment is handled through the :class:`~artiq.language.environment.HasEnvironment` manager provided by the :class:`~artiq.language.environment.EnvExperiment` class that experiments should derive from. 
 
 .. _device-db:
 
 The device database
 -------------------
 
-The device database contains information about the devices available in a ARTIQ installation, what drivers to use, what controllers to use and on what machine, and where the devices are connected.
+Information about available devices is provided to ARTIQ through a file called the device database. It is generated by a Python script typically called ``device_db.py``, which many of the ARTIQ front-end tools require access to in order to run. The device database specifies: 
```

Comment:
Generally while device_db files are initially generated (fact that it's a python script is irrelevant here imo), they can be modified by hand to suit whatever needs or experiments. I'd just focus on what it entails (like originally), rather than its origins.

Reply:
makes sense -- i'm generally trying to avoid *removing* information that was in the previous version of the manual, but i agree that this really doesn't add anything. i will change it together with the edits that will surely be brought up by sb :)